pkg/queue/stats.go

```diff
@@ -24,23 +24,16 @@ import (
 
 // ReportStats continually processes network events from reqCh and reports
 // aggregated stats via the `report` function whenever reportCh ticks.
-func ReportStats(startedAt time.Time, reqCh chan network.ReqEvent, reportCh <-chan time.Time, report func(float64, float64, float64, float64)) {
-	state := network.NewRequestStats(startedAt)
-
+func ReportStats(startedAt time.Time, stats *network.RequestStats, reportCh <-chan time.Time, report func(float64, float64, float64, float64)) {
 	for {
 		select {
-		case event, ok := <-reqCh:
-			if !ok {
```

Comment:
humorously this means we're back to this loop never exiting (which I personally think is fine but we did only put this block in 2 days ago or something :D)

Reply:
FWIW, I think we can remove this code soon'ish. I just didn't want to make the PR too big and that seems at least tangentially controversial as quite a few tests hang off of it.

Reply:
yep, was thinking the same thing - cool